packages/ui/src/components/va-image/VaImage.vue

```diff
@@ -1,122 +1,229 @@
 <template>
-  <div class="va-image" aria-live="polite">
-    <div
-      v-if="loadingError"
-      class="va-image__error"
+  <va-aspect-ratio v-bind="aspectRationAttributesComputed">
+    <picture
+      v-show="isSuccessfullyLoaded"
+      class="va-image"
+      :aria-busy="isLoading"
     >
-      <slot name="error" />
-    </div>
+      <slot v-if="$slots.sources" name="sources" />
 
-    <div :style="paddingStyles" />
-    <div
-      v-show="!loadingError && !loading"
-      class="va-image__img"
-    >
       <img
-        :style="imageStyles"
-        :src="$props.src"
-        :alt="$props.alt"
-        :draggable="$props.draggable"
+        ref="image"
+        v-bind="imgAttributesComputed"
         @error="handleError"
         @load="handleLoad"
       />
-    </div>
-    <div class="va-image__overlay">
+    </picture>
+
+    <div
+      v-if="$slots.default && isSuccessfullyLoaded"
+      class="va-image__overlay"
+    >
       <slot />
     </div>
+
+    <div
+      v-if="isError && $slots.error"
+      class="va-image__error"
+    >
+      <slot name="error" />
+    </div>
+
     <div
-      v-if="loading"
+      v-if="isLoading && $slots.loader"
       class="va-image__loader"
     >
       <slot name="loader" />
     </div>
-  </div>
+
+    <div
+      v-if="isPlaceholderShown"
+      class="va-image__placeholder"
+    >
+      <slot name="placeholder" />
+    </div>
+  </va-aspect-ratio>
 </template>
 
 <script lang="ts">
-import { defineComponent, computed, watch, ref } from 'vue'
-import { useComponentPresetProp } from '../../composables/useComponentPreset'
+import { defineComponent, ref, computed, watch, nextTick, onBeforeUnmount, type PropType, onBeforeMount } from 'vue'
+
+import { VaAspectRatio } from '../va-aspect-ratio'
+
+import {
+  useNativeImgAttributes, useNativeImgAttributesProps,
+  validateProp,
+} from './hooks/useNativeImgAttributes'
+import { useComponentPresetProp, useDeprecated } from '../../composables'
+
+const fitOptions = ['contain', 'fill', 'cover', 'scale-down', 'none'] as const
 
 export default defineComponent({
   name: 'VaImage',
+
+  components: { VaAspectRatio },
+
   emits: ['loaded', 'error'],
+
   props: {
     ...useComponentPresetProp,
-    ratio: { type: Number, default: 1 },
+    ...useNativeImgAttributesProps,
+    ratio: {
+      type: [Number, String] as PropType<number | 'auto'>,
+      default: 'auto',
+      validator: (v: number | 'auto') => {
+        if (typeof v === 'number') {
+          return v > 0
```

Comment:
I think `0` value will be invalid?

Reply:
Nope.
1. It will lead to zero divider.
2. It makes no sence because it should be width to height ratio.